### PR TITLE
`Development`: Add documentation for user experience

### PR DESCRIPTION
**Pull request by bassner**
#### General

- [x] I tested **all** changes and their related features with **all** corresponding user types on a test server.
- [x] Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://docs.artemis.ase.in.tum.de/dev/guidelines/language-guidelines/).
- [x] I chose a title conforming to the [naming conventions for pull requests](https://artemis-platform.readthedocs.io/en/latest/dev/guidelines/development-process.html#naming-conventions-for-github-pull-requests).

### Motivation and Context

Documentation for features supporting user satisfaction are currently missing.
See [the confluence page](https://confluence.ase.in.tum.de/display/ArTEMiS/Feature+documentation).

### Description

Add documentation pages for features that enhance the user experience:
- Guided Tours
- Theming

### Steps for Testing

- Proof read the new pages at https://artemis-platform--5368.org.readthedocs.build/en/5368/user/user-experience/
- Deploy the PR to a test server and check that the new link is present on the About Us page (but it will not work until merged, of course)

#### Text Review
- [x] Review 1
- [x] Review 2

### CODE REVIEW

**Review by luisa-we — COMMENTED** [review]
Have read through the documentation, easy to understand and the videos are very helpful.
Tested on TS2, the new link shows up in the English, but not in the German version. Maybe you could add the link there too?

Bildschirmfoto 2022-07-11 um 23 02 04

**Comment by bassner** [comment]
Oops - thanks for the heads up, @luisa-we! Added in 555565e.

**Review by luisa-we — DISMISSED** [review]
Tested on TS2, the link is now also included in the German version!

File: docs/user/user-experience/themes.rst [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+You can change the theme manually at any time by clicking the theme switcher icon in the navbar:
+
+.. figure:: ./theme_switch.gif
+    :alt: Click the icon in the navbar to change the theme
+
+Synchronize the theme with your operating system
+------------------------------------------------
+
+By default, Artemis will choose the dark theme if you indicated that you prefer dark themes in your OS settings.
```
Comment (Strohgelaender) [inline docs/user/user-experience/themes.rst:23]:
```suggestion
By default, Artemis will choose the preferred theme according to your OS settings.
```

**Review by jschmautzer — COMMENTED** [review]
only two small things I noticed in the documentation:
Under the section Guided Tours > Restart or resume Guided Tours in the Note it says "You need to be on the page where you started the tour the last time resume or restart it." Shouldn't it be something like: "You need to be on the page where you started the tour the last time **to** resume or restart it."
And one other point in the section User Interface Themes > Synchronize the theme with your operating system: In the last sentence, it maybe would be more clear, if you add something like: "In the following example, the user .... , so the theme changes automatically, **when enabling the synchronized behavior.**"

File: docs/user/user-experience/guided_tours.rst [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+is available on your current page by opening your account menu and selecting it:
+
+.. figure:: ./resume_tutorial.gif
+    :alt: Open user menu to resume a tour
+
+You can run all tours as often as you'd like. Any changes to the tutorial course you performed during your last run
+will be reset automatically.
+
+.. note::
+    You need to be on the page where you started the tour the last time resume or restart it. The menu will
```
[the remark this answers is not in the document]
Comment (bassner) [reply docs/user/user-experience/guided_tours.rst]: [9e7288f](https://github.com/ls1intum/Artemis/pull/5368/commits/9e7288f40af7c2a0f8c7f94307ddc30d17f69a74)

File: docs/user/user-experience/guided_tours.rst [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+| Each of the following steps will ask you either to perform a certain step or to watch / read additional information.
+| Please follow any instructions presented to you.
+
+You can exit a Guided Tour at any time by clicking the close-icon in the top right of the modal box.
+
+Restart or resume Guided Tours
+------------------------------
+
+In case you have cancelled a previous tour or want to do an already finished tour again, you can start any tour that
+is available on your current page by opening your account menu and selecting it:
```
[the remark this answers is not in the document]
Comment (bassner) [reply docs/user/user-experience/guided_tours.rst]: [9e7288f](https://github.com/ls1intum/Artemis/pull/5368/commits/9e7288f40af7c2a0f8c7f94307ddc30d17f69a74)

File: docs/user/user-experience/guided_tours.rst [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+.. figure:: ./tour_example.gif
+    :alt: Course overview Guided Tour
+
+| To advance with the tour, click the *Next* button. You can go back to a previous step using the *Back* button.
+| Each of the following steps will ask you either to perform a certain step or to watch / read additional information.
+| Please follow any instructions presented to you.
+
+You can exit a Guided Tour at any time by clicking the close-icon in the top right of the modal box.
+
+Restart or resume Guided Tours
```
[the remark this answers is not in the document]
Comment (bassner) [reply docs/user/user-experience/guided_tours.rst]: [9e7288f](https://github.com/ls1intum/Artemis/pull/5368/commits/9e7288f40af7c2a0f8c7f94307ddc30d17f69a74)

File: docs/user/user-experience/themes.rst (hunk 1 above)
Comment (bassner) [reply docs/user/user-experience/themes.rst]: 244f84

**Comment by bassner** [comment]
Hey @jschmautzer 
Thanks for your comment
The first point was addressed in @Strohgelaender's review above :)
I've added your 2nd suggestions slightly modified in 012bb29

**Review by jbllhff — APPROVED** [review]
With the suggested changes, this looks good. Great use of the video tutorials!

**Review by Strohgelaender — APPROVED** [review]

**Review by MetinLamby — COMMENTED** [review]
very clear documentation. Just one comment regarding consistency: 
if you look at all other documentation pages of the user guide, the actual documentation is on the same page as the table of content (eg. https://artemis-platform--5367.org.readthedocs.build/en/5367/user/scaling/) 

In the UX documentation, the actual content of the documentation cannot be accessed unless you click on the links of the table of content which is not the case for the other documentations. Maybe for consistency, we can agree on one format for displaying table of content and the content of the documentation

**Review by jschmautzer — APPROVED** [review]
documentation is understandable :) 
also tested on ts2, worked as described

**Review by Mikael31 — COMMENTED** [review]
Videos all run smoothly.
I don't know if it would be good to have the option to restart or pause the video.
Other than that it looked consistent and correct.

**Comment by bassner** [comment]
> Videos all run smoothly. I don't know if it would be good to have the option to restart or pause the video. Other than that it looked consistent and correct.

They are GIFs, so no way to pause them! :D

**Comment by bassner** [comment]
@MetinLamby I've taken the style from the exam mode where the table of contents is on the main page and only links to subpages https://artemis-platform--5368.org.readthedocs.build/en/5368/user/exam_mode/

**Review by Mikael31 — APPROVED** [review]
Then everything is approved from my side

**Review by Marvin-Vitcu — APPROVED** [review]
Tested on TS1. Works as described.

**Pull request merged**